src/Language/PureScript/CoreFn/Desugar.hs

```diff
@@ -32,7 +32,7 @@ moduleToCoreFn :: Environment -> A.Module -> Module Ann
 moduleToCoreFn _ (A.Module _ _ _ _ Nothing) =
   internalError "Module exports were not elaborated before moduleToCoreFn"
 moduleToCoreFn env (A.Module modSS coms mn decls (Just exps)) =
-  let imports = mapMaybe importToCoreFn decls ++ fmap (ssAnn modSS,) (findQualModules decls)
+  let imports = fmap (ssAnn modSS,) (findQualModules decls) ++ mapMaybe importToCoreFn decls
```

Comment:
Why was this change necessary?

Reply:
`imports` is a list of all imports for the module, be it the ones that were explicitly declared on the source file (the `mapMaybe importToCoreFn decls` part) or inferred ones from internal modules. 
They are deduplicated soon after this piece of code, so only one set of annotations remain after the deduplication. 
The dedupe code uses `Map.fromListWith`, which runs from right to left and keeps the unique values. So the values we want to keep (annotations declared on the source code) need to be added to this list later so that they're taking into consideration first by `Map.fromListWith`.